src/pages/settings/company/edit/CompanyEdit.tsx

```diff
@@ -0,0 +1,205 @@
+/**
+ * Invoice Ninja (https://invoiceninja.com).
+ *
+ * @link https://github.com/invoiceninja/invoiceninja source repository
+ *
+ * @copyright Copyright (c) 2022. Invoice Ninja LLC (https://invoiceninja.com)
+ *
+ * @license https://www.elastic.co/licensing/elastic-license
+ */
+
+import { Card, Element } from '@invoiceninja/cards';
+import { Button, InputField, SelectField } from '@invoiceninja/forms';
+import { AxiosError } from 'axios';
+import { endpoint } from 'common/helpers';
+import { request } from 'common/helpers/request';
+import { route } from 'common/helpers/route';
+import { toast } from 'common/helpers/toast/toast';
+import { useCurrentCompany } from 'common/hooks/useCurrentCompany';
+import { useLanguages } from 'common/hooks/useLanguages';
+import { CompanyInput } from 'common/interfaces/company-interface';
+import { ValidationBag } from 'common/interfaces/validation-bag';
+import { CurrencySelector } from 'components/CurrencySelector';
+import { LightSwitch } from 'components/LightSwitch';
+import { Modal } from 'components/Modal';
+import {
+  FormEvent,
+  useState,
+  SetStateAction,
+  Dispatch,
+  useEffect,
+} from 'react';
+import { useTranslation } from 'react-i18next';
+
+interface Props {
+  isModalOpen: boolean;
+  setIsModalOpen: Dispatch<SetStateAction<boolean>>;
+}
+
+export function CompanyEdit(props: Props) {
+  const [t] = useTranslation();
+
+  const languages = useLanguages();
+
+  const currentCompany = useCurrentCompany();
+
+  const [errors, setErrors] = useState<ValidationBag>();
+
+  const [isFormBusy, setIsFormBusy] = useState<boolean>(false);
+
+  const [companyId, setCompanyId] = useState<string | undefined>();
+
+  const [company, setCompany] = useState<CompanyInput>();
+
+  const fetchCompanyDetails = async (company_id: string) => {
```

Comment:
@beganovich The reason I made a data fetch like this instead of using hooks is because we have a different company_ids on the mount of this component. First render -> undefined, Second render -> ID of the first company in the list, Third render -> ID of the actual company we selected. So if I use a hook instead of a function like this on the second render I will get a 401 error because I'm trying to access a currently unselected company.